noir-projects/aztec-nr/aztec/src/messages/discovery/process_message.nr

```diff
@@ -0,0 +1,70 @@
+use crate::messages::{
+    discovery::{
+        ComputeNoteHashAndNullifier, partial_notes::process_partial_note_private_msg,
+        private_events::process_private_event_msg, private_notes::process_private_note_msg,
+    },
+    encoding::{decode_message, MAX_MESSAGE_LEN},
+    msg_type::{
+        PARTIAL_NOTE_PRIVATE_MSG_TYPE_ID, PRIVATE_EVENT_MSG_TYPE_ID, PRIVATE_NOTE_MSG_TYPE_ID,
+    },
+    processing::message_context::MessageContext,
+};
+
+use protocol_types::{address::AztecAddress, debug_log::{debug_log, debug_log_format}};
+
+/// Processes a message that can contain notes, partial notes, or events.
+///
+/// Notes result in nonce discovery being performed prior to delivery, which requires knowledge of the transaction hash
+/// in which the notes would've been created (typically the same transaction in which the log was emitted), along with
+/// the list of unique note hashes in said transaction and the `compute_note_hash_and_nullifier` function. Once
+/// discovered, the notes are enqueued for validation.
+///
+/// Partial notes result in a pending partial note entry being stored in a PXE capsule, which will later be retrieved to
+/// search for the note's completion public log.
+///
+/// Events are processed by computing an event commitment from the serialized event data and its randomness field, then
+/// enqueueing the event data and commitment for validation.
+pub unconstrained fn do_process_message<Env>(
+    contract_address: AztecAddress,
+    compute_note_hash_and_nullifier: ComputeNoteHashAndNullifier<Env>,
+    message: BoundedVec<Field, MAX_MESSAGE_LEN>,
+    message_context: MessageContext,
+) {
```

Comment:
The body of this function is basically a copy-paste from `process_private_log`.

Note that it seems to me a bit a strange that function called `do_process_message` resides in the `discovery` directory when we have `processing` directory right next to it. Would it be preferable to have this in `processing`?

Reply:
`processing` initially referred to other things, but yes, this should likely fall there. Let's not rename it here however, since I have some open branches that move some things around. Better to have renamings be separate from changes.

Reply:
Ok, will not do anything in this PR with the intention of this eventually being addressed.